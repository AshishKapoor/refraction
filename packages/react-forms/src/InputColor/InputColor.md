The `InputColor` component shares the same API of the HTML `input[type=color]` element.

```js
<InputColor onChange={e => console.log(e.target.value)} />
```

It also supports all the `InputText` and `Button` properties:

```js
<InputColor size="small" defaultValue="#FFF000" importance="primary" />
```

The component supports both uncontrolled and controlled API styles:

```js
initialState = { value: '#00FF00' };
<InputColor
  size="large"
  value={state.value}
  onChange={evt => setState({ value: evt.target.value })}
/>;
```
