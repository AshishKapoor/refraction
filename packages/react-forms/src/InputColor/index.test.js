/**
 * Copyright (c) Quid, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
// @flow
import React from 'react';
import { mount } from 'enzyme';
import InputColor from './index';

jest.mock('../InputText', () => ({
  __esModule: true,
  default: 'input',
  HEIGHT: jest.requireActual('../InputText').HEIGHT,
}));
jest.mock('../Button', () => 'button');
jest.mock('@quid/react-core', () => ({
  Icon: 'x-icon',
}));

const noop = () => undefined;

it('renders the expected markup', () => {
  const wrapper = mount(<InputColor onChange={noop} />);

  expect(wrapper).toMatchSnapshot();
});

it('responds to user interaction on fake button', () => {
  const wrapper = mount(<InputColor onChange={noop} />);

  wrapper.find('input[type="color"]').simulate('mouseEnter');
  wrapper.find('input[type="color"]').simulate('mouseDown');
  wrapper.find('input[type="color"]').simulate('focus');

  expect(wrapper.find('button').prop('data-state')).toMatchInlineSnapshot(
    `"hover active focus"`
  );

  wrapper.find('input[type="color"]').simulate('mouseLeave');
  wrapper.find('input[type="color"]').simulate('mouseUp');
  wrapper.find('input[type="color"]').simulate('blur');

  expect(wrapper.find('button').prop('data-state')).toMatchInlineSnapshot(`""`);
});

it('renders a disabled component', () => {
  const wrapper = mount(<InputColor disabled onChange={noop} />);

  expect(
    wrapper
      .find('input')
      .at(0)
      .prop('disabled')
  ).toBe(true);
});

it('triggers onChange when color is changed using the `type=color` input', () => {
  const handleChange = jest.fn();
  const wrapper = mount(<InputColor onChange={handleChange} />);

  wrapper
    .find('input')
    .at(1)
    .simulate('change', { target: { value: '#000000' } });

  expect(handleChange.mock.calls[0][0]).toMatchObject({
    target: { value: '#000000' },
  });
});

it('triggers onChange when color is changed using the `type=text` input', () => {
  const handleChange = jest.fn();
  const wrapper = mount(<InputColor onChange={handleChange} />);

  wrapper
    .find('input')
    .at(0)
    .simulate('change', { target: { value: '#000000' } });

  expect(handleChange.mock.calls[0][0]).toMatchObject({
    target: { value: '#000000' },
  });
});
