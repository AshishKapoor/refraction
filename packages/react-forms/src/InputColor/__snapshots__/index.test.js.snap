// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the expected markup 1`] = `
.emotion-7 {
  all: unset;
  display: inline-block;
  border-radius: 2px;
  -webkit-transition: 0.2s ease-in-out border-color;
  transition: 0.2s ease-in-out border-color;
  border: 1px solid #8F9BA3;
}

.emotion-7:focus-within {
  border-color: #00c1bb;
}

.emotion-7[data-invalid='true'] {
  border-color: #E61E27;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  height: 32px;
}

.emotion-1 {
  opacity: 0;
  position: absolute;
  width: 100%;
  height: 100%;
  overflow: hidden;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  cursor: pointer;
}

.emotion-1:focus:not(:focus-visible) ~ button {
  box-shadow: none;
}

.emotion-0 {
  border-radius: 1px 0 0 1px;
  border-left: 0;
  border-top: 0;
  border-bottom: 0;
}

.emotion-3 {
  border-radius: 0 1px 1px 0;
  border-right: 0;
  border-top: 0;
  border-bottom: 0;
}

<InputColor
  onChange={[Function]}
>
  <ForwardRef
    className="emotion-9 emotion-6"
    onChange={[Function]}
  >
    <InputGroup
      className="emotion-9 emotion-6"
    >
      <fieldset
        className="emotion-6 emotion-7 emotion-8"
        data-invalid={false}
        onChange={[Function]}
        onInvalid={[Function]}
      >
        <input
          className="emotion-0"
          key=".0"
          maxLength={7}
          onChange={[Function]}
          value="#000000"
        />
        <Wrapper
          key=".1"
        >
          <div
            className="emotion-4 emotion-5"
          >
            <ColorSelector
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              type="color"
              value="#000000"
            >
              <input
                className="emotion-1 emotion-2"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                type="color"
                value="#000000"
              />
            </ColorSelector>
            <button
              className="emotion-3"
              data-state=""
              importance="secondary"
              tabIndex={-1}
            >
              <x-icon
                name="tint"
                style={
                  Object {
                    "color": "#000000",
                  }
                }
              />
            </button>
          </div>
        </Wrapper>
      </fieldset>
    </InputGroup>
  </ForwardRef>
</InputColor>
`;
